File: app/src/main/java/com/lzh/nonview/router/demo/action/SayHelloAction.java
package com.lzh.nonview.router.demo.action;

import android.content.Context;
import android.os.Bundle;
import android.widget.Toast;

import com.lzh.nonview.router.anno.RouterRule;
import com.lzh.nonview.router.route.ActionSupport;

@RouterRule({"haoge://haoge.cn/hello"})
public class SayHelloAction extends ActionSupport {
    @Override
    public void onRouteTrigger(Context context, Bundle bundle) {
        Toast.makeText(context, "Hello", Toast.LENGTH_SHORT).show();
    }
}

File: sample/src/main/java/com/lzh/nonview/router/demo/action/SayHelloAction.kt
package com.lzh.nonview.router.demo.action

import android.content.Context
import android.os.Bundle
import com.haoge.easyandroid.easy.EasyToast
import com.lzh.nonview.router.anno.RouterRule
import com.lzh.nonview.router.route.ActionSupport

/**
 * @author haoge on 2018/5/10
 */
@RouterRule("say/hello")
class SayHelloAction:ActionSupport() {
    override fun onRouteTrigger(context: Context?, bundle: Bundle?) {
        EasyToast.DEFAULT.show("Hello! this is an action route!")
    }
}